R/migrate.project.R

```diff
@@ -17,118 +17,127 @@
 #' \dontrun{migrate.project()}
 migrate.project <- function()
 {
-  my.project.info <- list()
-
   message('Migrating project configuration')
 
   # Load the config and look for specific problems in the configuration that
   # should be fixed (e.g. missing files, missing config item)
   # Also flag up if any items need special handling during migration (for example
   # if something other than the default is appropriate for existing projects)
-  
+
   # Initialise migration flags
   config_conflicts <- FALSE
+  directory_conflicts <- FALSE
   config_warnings <- NULL
-  
-  
+
   # Flags stored in environment env
   env <- environment()
-  
+
   # Detect any conflicts with the existing config file once it has been processed
   # during load.project() (flag for now and handle later on)
-  
-  loaded.config <- tryCatch(.load.config(),
-                      warning=function(w) {
-                          # set up some variables to help process the 
-                          # migration warnings later
-                          
-                          assign("config_conflicts", TRUE, envir = env)
-                          assign("config_warnings", w$message, envir = env)
-                          suppressWarnings(.load.config())
-                      })
-          
-  # Detect other migration issues 
-  
-  
+  loaded.config <- tryCatch(
+    .load.config(),
+    warning = function(w) {
+      # set up some variables to help process the
+      # migration warnings later
+
+      assign("config_conflicts", TRUE, envir = env)
+      assign("config_warnings", w$message, envir = env)
+      suppressWarnings(.load.config())
+    }
+  )
+
+  # Detect other migration issues
+  doc_not_renamed <- dir.exists("doc") && !dir.exists("docs")
+  directory_conflicts <- any(doc_not_renamed)
```

Comment:
I was only meaning that `docs_directory_conflicts` is more specific than `directory_conflicts` so it might be more descriptive (ie, it's not a conflict to do with say the `reports` directory).

I prefer the way you have coded the if statement because each conflict needs a specific migration action, so having a more general flag wouldn't help.